### PR TITLE
Apply automatic symfony 4.4 upgrades

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no 
| New feature? | no 
| BC breaks? | no 
| Deprecations? | no  
| Fixed tickets | fixes # 
| Related issues/PRs | # 
| License | MIT
| Documentation PR | sulu/sulu-docs# 

#### What's in this PR?

 - Apply automatic Symfony 4.4 upgrades from [rector](https://github.com/rectorphp/rector).
   - Add missing command return type
   - Migrate event listener to event subscribers
 - Manual fix not found class issues for rector

```bash
composer require rector/rector --dev
```

`rector.php`

```php
<?php

declare(strict_types=1);

/*
 * This file is part of Sulu.
 *
 * (c) Sulu GmbH
 *
 * This source file is subject to the MIT license that is bundled
 * with this source code in the file LICENSE.
 */

use Rector\Core\Configuration\Option;
use Rector\PHPUnit\Set\PHPUnitSetList;
use Rector\Symfony\Set\SymfonySetList;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function(ContainerConfigurator $containerConfigurator): void {
    // region Symfony Container
    $parameters = $containerConfigurator->parameters();
    $parameters->set(
        Option::SYMFONY_CONTAINER_XML_PATH_PARAMETER,
        __DIR__ . '/var/cache/admin/dev/App_KernelDevDebugContainer.xml'
    );
    $parameters->set(
        Option::SKIP,
        [
            __DIR__ . '/src/Sulu/Bundle/*/Tests/Application/var/*',
            __DIR__ . '/src/Sulu/Bundle/TestBundle/Tests/app/var/*',
            __DIR__ . '/src/Sulu/Bundle/*/Resources/phpcr-migrations/*',
            __DIR__ . '/src/Sulu/Bundle/*/DependencyInjection/Compiler/*',
            __DIR__ . '/src/Sulu/Bundle/TestBundle/Testing/*Trait.php',
        ]
    );
    // endregion

    $containerConfigurator->import(SymfonySetList::SYMFONY_44);
    $containerConfigurator->import(SymfonySetList::SYMFONY_CODE_QUALITY);
    $containerConfigurator->import(SymfonySetList::SYMFONY_CONSTRUCTOR_INJECTION);
    $containerConfigurator->import(PHPUnitSetList::PHPUNIT_80);
};
```

```bash
vendor/bin/rector process src
```

#### Why?

Keep things sulu up to date.

#### To Do

- [ ] Create a documentation PR
- [ ] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php [hunk 1]
```diff
@@ -39,6 +40,11 @@ public function __construct(
         $this->searchManager = $searchManager;
     }
 
+    public static function getSubscribedEvents(): array
+    {
+        return ['sulu_security.permission_update' => 'onPermissionUpdate'];
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php]: isnt there a constant for this?

File: src/Sulu/Bundle/PageBundle/Search/EventListener/PermissionListener.php [hunk 2]
```diff
@@ -37,6 +38,11 @@ public function __construct(DocumentManagerInterface $documentManager, SearchMan
         $this->searchManager = $searchManager;
     }
 
+    public static function getSubscribedEvents(): array
+    {
+        return ['sulu_security.permission_update' => 'onPermissionUpdate'];
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/PageBundle/Search/EventListener/PermissionListener.php]: isnt there a constant for this?

File: src/Sulu/Component/Content/Import/WebspaceImportInterface.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ interface WebspaceImportInterface
      * @param bool $overrideSettings
      * @param string $exportSuluVersion
      *
-     * @return array
+     * @return \stdClass
```
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Import/WebspaceImportInterface.php:33]: are you sure that this change is correct?

**Review by wachterjohannes — CHANGES_REQUESTED** [review]

File: src/Sulu/Component/Content/Import/WebspaceImportInterface.php (hunk 3 above)
Comment (alexander-schranz) [reply src/Sulu/Component/Content/Import/WebspaceImportInterface.php:33]: yes

File: src/Sulu/Bundle/PageBundle/Search/EventListener/PermissionListener.php (hunk 2 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/PageBundle/Search/EventListener/PermissionListener.php]: Good catch didn't found first as the constant was in `Sulu\Component` namespace 🙈

File: src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Search/EventListener/PermissionListener.php]: fixed

**Review by wachterjohannes — APPROVED** [review]

**Pull request merged**